app/src/main/java/com/nmakademija/nmaakademija/fragment/NewsFragment.java

```diff
@@ -42,8 +42,8 @@ public void onActivityCreated(@Nullable Bundle savedInstanceState) {
             public void onResponse(Call<List<Article>> call, Response<List<Article>> response) {
                 View v = getView();
                 if (v != null) {
-                    RecyclerView rv = (RecyclerView) getView().findViewById(R.id.recyclerView);
-                    rv.setAdapter(new ArticlesAdapter(response.body()));
+                    ListView listView = (ListView) getView().findViewById(R.id.recyclerView);
+                    listView.setAdapter(new ArticlesAdapter(getContext(), (ArrayList<Article>) response.body()));
```

Comment:
Change `ArticlesAdapter` constructor to take `List<Article>`. If possible avoid such casts